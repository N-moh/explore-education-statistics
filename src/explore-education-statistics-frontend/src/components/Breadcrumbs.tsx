import React from 'react';
import { Link, Route } from 'react-router-dom';

interface Props {
  current: string;
}

const Breadcrumbs = ({ current }: Props) => (
  <Route path="*">
    <div className="govuk-breadcrumbs">
      <ol className="govuk-breadcrumbs__list">
        <li className="govuk-breadcrumbs__list-item">
          <Link className="govuk-breadcrumbs__link" to="/">
            Home
          </Link>
        </li>
        {window.location.pathname
          .split('/')
          .slice(1)
          .map((elem, index) => (
            <li key={index} className="govuk-breadcrumbs__list-item">
              <Link className="govuk-breadcrumbs__link" to={`/${elem}`}>
                {elem}
              </Link>
            </li>
          ))}
      </ol>
    </div>
  </Route>
);

export default Breadcrumbs;
