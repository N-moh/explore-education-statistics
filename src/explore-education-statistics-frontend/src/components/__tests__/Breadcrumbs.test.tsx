import React from 'react';
import { MemoryRouter } from 'react-router';
import { Route } from 'react-router-dom';
import { render } from 'react-testing-library';
import Breadcrumbs from '../Breadcrumbs';

describe('Breadcrumbs', () => {
  test('renders correctly with just home breadcrumb', () => {
    const path = '/';

    const { container } = render(
      <MemoryRouter initialEntries={[path]}>
        <Route path={path} exact>
          <Breadcrumbs />
        </Route>
      </MemoryRouter>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('renders correctly with multiple breadcrumbs', () => {
    const path = '/publications/test-publication';

    const { container } = render(
      <MemoryRouter initialEntries={[path]}>
        <Route path={path} exact>
          <Breadcrumbs />
        </Route>
      </MemoryRouter>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });
});
