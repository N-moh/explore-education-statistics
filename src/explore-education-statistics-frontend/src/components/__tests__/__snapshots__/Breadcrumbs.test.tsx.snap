// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders correctly with just home breadcrumb 1`] = `
<div class="govuk-breadcrumbs">
  <ol class="govuk-breadcrumbs__list"
      data-testid="breadcrumbs--list"
  >
    <li class="govuk-breadcrumbs__list-item">
      <a role="link"
         tabindex="-1"
         class="govuk-link govuk-breadcrumbs__link"
         href="/"
      >
        Home
      </a>
    </li>
  </ol>
</div>
`;

exports[`Breadcrumbs renders correctly with multiple breadcrumbs 1`] = `
<div class="govuk-breadcrumbs">
  <ol class="govuk-breadcrumbs__list"
      data-testid="breadcrumbs--list"
  >
    <li class="govuk-breadcrumbs__list-item">
      <a role="link"
         tabindex="-1"
         class="govuk-link govuk-breadcrumbs__link"
         href="/"
      >
        Home
      </a>
    </li>
    <li class="govuk-breadcrumbs__list-item">
      <a role="link"
         tabindex="-1"
         class="govuk-link govuk-breadcrumbs__link"
         href="/publications"
      >
        Publications
      </a>
    </li>
    <li class="govuk-breadcrumbs__list-item">
      Test publication
    </li>
  </ol>
</div>
`;
